string.c

```diff
@@ -5804,16 +5804,14 @@ rb_str_aref(VALUE str, VALUE indx)
 
 /*
  *  call-seq:
+ *    self[substring] -> new_string or nil
+ *    self[regexp, capture = 0] -> new_string or nil
  *    self[index] -> new_string or nil
  *    self[start, length] -> new_string or nil
  *    self[range] -> new_string or nil
- *    self[regexp, capture = 0] -> new_string or nil
- *    self[substring] -> new_string or nil
  *
```

Comment:
What's the reason to change the order here? It seems like the original order was in the order which this method is most commonly used, which I think we should keep (and also change the method documentation to follow it).